code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -1023,19 +1023,61 @@
 /datum/reagent/minttoxin
 	name = "Mint Toxin"
 	id = MINTTOXIN
-	description = "Useful for dealing with undesirable customers."
+	description = "Useful for dealing with undesirable customers. The undiluted version of Mint Extract."
 	reagent_state = REAGENT_STATE_LIQUID
 	color = "#CF3600" //rgb: 207, 54, 0
 	density = 0.898
 	specheatcap = 3.58
+	custom_metabolism = 0.01 //so it lasts 10x as long as regular minttox
+	var/fatgokaboom = TRUE
+	nutriment_factor = 2.5 * REAGENTS_METABOLISM //about as nutritious as sugar
+	sport = SPORTINESS_SUGAR //a small performance boost from being COOL AND FRESH
+	var/chillcounter = 0
 
 /datum/reagent/minttoxin/on_mob_life(var/mob/living/M, var/alien)
 
 	if(..())
 		return 1
 
-	if(M_FAT in M.mutations)
+	if(prob(5))
+		to_chat(M, "<span class='notice'>[pick("You feel minty fresh!","If freshness could kill you'd be a serial killer!","You feel the strange urge to share this minty freshness with others!","You have a sudden craving to drink ice cold water.","Ahh, so refreshing!")]</span>")
+
+	if(M.bodytemperature > 310) //copypasted from the cold drinks check so I don't have to change minttox internally and maybe most certainly break shit in the process
+		M.bodytemperature = max(310, M.bodytemperature + (-5 * TEMPERATURE_DAMAGE_COEFFICIENT)) //that minty freshness my dude, chill out
+
+	if(fatgokaboom && M_FAT in M.mutations)
 		M.gib()
+		
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(holder.has_reagent(COLDDRINKS) & prob(10))
+			var/datum/butchering_product/teeth/J = locate(/datum/butchering_product/teeth) in H.butchering_drops
+			if(J.amount == 0)
+				return
+			else
+				H.custom_pain(pick("AHHH YOUR TEETH HURT!","You didn't know you had a cavity. You do now.","DAMN YOUR TEETH HURT"),5)
+				holder.reagents.add_reagent(SACID,1) //just a smidgeon
```

Comment:
holder _is_ the reagents datum

please compile your code unless you are explicitly posting as a WIP!

Reply:
I'm was jury rigging it while doing something. I do check the compile fail errors, but again, I don't really understand byondcode enough to not use extensive trial and error.

Reply:
And I can't really compile it locally, since there's some code on the repo that's for 515, it breaks on my local copy (which is still on stable). The solution for that is back and forth installing of the beta and stable, but I don't really wanna do that, so I use the github compile test.

Reply:
>installing

no need to do that, king
![image](https://github.com/vgstation-coders/vgstation13/assets/18052467/471b1852-33e3-49f7-ad6b-7152fbcd342d)

just download the 515 zip and run the stuff from bin(make sure you dont have a 514 dreamdaemon already running)

Reply:
You know, I had totally forgotten a zip file was available. Lessons for the future I guess.